spec/controllers/idv/hybrid_handoff_controller_spec.rb

```diff
@@ -244,6 +253,36 @@
         end
       end
     end
+
+    context 'with selfie enabled system wide' do
+      let(:doc_auth_selfie_capture_enabled) { true }
+      before do
+        # allow(controller).to receive(:current_sp).
+        #   and_return(service_provider)
+      end
+      describe 'when selfie is enabled for sp' do
+        let(:sp_selfie_enabled) { true }
+        it 'pass on correct flags and states and logs correct info' do
+          allow(controller).to receive(:sp_session).
+            and_return({ biometric_comparison_required: sp_selfie_enabled })
+          get :show
+          expect(response).to render_template :show
+          expect(subject.idv_session.selfie_check_required).to eq(true)
+          expect(@analytics).to have_logged_event(analytics_name, analytics_args)
+        end
+      end
+      describe 'when selfie is disabled for sp' do
+        let(:sp_selfie_enabled) { false }
+        it 'pass on correct flags and states and logs correct info' do
+          allow(controller).to receive(:sp_session).
+            and_return({ biometric_comparison_required: sp_selfie_enabled })
```

Comment:
```suggestion
          allow(controller.decorated_sp_session).to receive(:selfie_required?).
            and_return(sp_selfie_enabled)
```

Reply:
sounds good ... it woud work either way since the user who does not see the Hybrid Handoff is on mobile  and the precondition clause takes that into consideration 

@dawei-nava there's actually an edge case  I remembered of which of which I have concern ... an SP can change during IdV meaning the selfie check requirement can change midway through IdV. when I initially prototyped this (not in the prototype I shared with you) i created a before filter in the document capture controller that set the idv_session.selfie_check_required value for use in the preconditions Proc.  I'm thinking that this may be best path forward to avoid this "edge" case that could be more common than I think.

Reply:
@amirbey , i noticed following, i think it's the edge case you mentioned?

https://github.com/18F/identity-idp/blob/877a294f3a442fa1eb726ea8e49db1809d73fd03/app/controllers/concerns/idv_session_concern.rb#L73-L75

Reply:
@amirbey , not so sure you say selfie requirement can change through idv, do you mean the sp changed it's configuration like what's in `service_providers.yml` dictates?

If so, the situation sounds like when an user is shopping on a website with merchandizes in the shopping cart. Do we actively force the system to recheck the prices of each item in the cart? Either way the risk still exists that something's price changed between last check and order saved in the system. So we have to ask what's the chance of this happens.

To strictly prevent this, the system has to lock the SP configuration when there is session for that SP is in progress.

Reply:
The concern is not a change in the service_provider yaml.  But if a user attempts to login to a different SP midway during IdV, the sp_session can change on the next request to the backend.

Reply:
@amirbey, then it may be better to query it every time `decorated_sp_session.selfie_required?` other than bookkeeping the state in idv session.

Reply:
@amirbey , changed based on discussion and clarification.